modules/operations/cycle_highway_tag.js

```diff
@@ -6,31 +6,62 @@ import { KeyOperationBehavior } from '../behaviors/KeyOperationBehavior';
 let _wasSelectedIDs = [];
 let _wasPresetIDs = [];
 
-
 export function operationCycleHighwayTag(context, selectedIDs) {
   const editor = context.systems.editor;
   const graph = editor.staging.graph;
   const l10n = context.systems.l10n;
   const presets = context.systems.presets;
 
   // Allow cycling through lines that match these presets
-  const allowPresetRegex = [
+  const allowHighwayPresetRegex = [
     /^highway\/(motorway|trunk|primary|secondary|tertiary|unclassified|residential|living_street|service|track)/,
-    /^line$/
+    /^line$/,
   ];
 
-  const defaultPresetIDs = [
+  const defaultHighwayPresetIDs = [
     'highway/residential',
     'highway/service',
     'highway/track',
     'highway/unclassified',
     'highway/tertiary',
-    'line'
+    'line',
+  ];
+
+  // Allow cycling through crossings that match these presets
+  const allowCrossingPresetRegex = [
+    /^crossing\/(unmarked|marked|marked:zebra|marked:lines|marked:dashes|marked:ladder|marked:dots|marked:ladder:skewed)/,
+  ];
+
+  const defaultCrossingPresetIDs = [
+    'crossing/unmarked;crossing:markings=no',
+    'crossing/marked;crossing:markings=yes',
+    'crossing/marked;crossing:markings=zebra',
+    'crossing/marked;crossing:markings=lines',
+    'crossing/marked;crossing:markings=ladder',
+    'crossing/marked;crossing:markings=dashes',
+    'crossing/marked;crossing:markings=dots',
+    'crossing/marked;crossing:markings=ladder:skewed',
   ];
 
-  // same selection as before?
-  const isSameSelection = utilArrayIdentical(selectedIDs, _wasSelectedIDs);
-  const presetIDs = new Set(isSameSelection ? _wasPresetIDs : defaultPresetIDs);
+  // Check if selection is highway or crossing
+  const isCrosswalkSelection = selectedIDs.some((id) => id.includes('crosswalk'));
+  const isHighwaySelection = selectedIDs.some((id) => id.includes('highway'));
+
```

Comment:
Ahhhh you are gettting really close. The code you've written is considering multi-selection (selecetedIDs is an array after all) but let's just consider single selections. 

Replace these lines with: 

```
  //Do not allow multi-select.
  if (selectedIDs.length > 1) return false;

  const selectedID = selectedIDs[0];

  const entity = graph.hasEntity(selectedID);

  // Check if selection is highway or crossing
  const isCrosswalkSelection = !!entity.tags.crosswalk;
  const isHighwaySelection = !entity.tags.crosswalk && !!entity.tags.highway;
```

Reply:
Thank you, @Bonkles, for the review. I will make the changes.